### PR TITLE
Normalize slashes for paths in watch events

### PR DESCRIPTION
**Pull request by sheetalkamat**

### CODE REVIEW

**Review by jakebailey — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
@typescript-bot cherry-pick to release-5.4

File: src/testRunner/unittests/tsserver/events/watchEvents.ts [hunk 1]
```diff
@@ -93,8 +93,8 @@ describe("unittests:: tsserver:: events:: watchEvents", () => {
     function updateFileOnHost(session: TestSession, file: string, log: string, content?: string) {
         // Change b.ts
         session.logger.log(`${log}: ${file}`);
-        if (content) session.host.appendFile(file, content);
-        else session.host.writeFile(file, session.host.readFile("/user/username/projects/myproject/a.ts")!);
+        if (content && session.host.fileExists(file)) session.host.appendFile(file, content);
```
Comment (DanielRosenwasser) [inline src/testRunner/unittests/tsserver/events/watchEvents.ts:96]: I know this is just testing, but is there technically an issue here with if `content` is an empty string?

Comment (sheetalkamat) [reply src/testRunner/unittests/tsserver/events/watchEvents.ts:96]: yeah but its for testing so just simplified it

**Comment by sheetalkamat** [comment]
@DanielRosenwasser conflict for picking up comes because https://github.com/microsoft/TypeScript/pull/57967 is not merged

**Comment by DanielRosenwasser** [comment]
@typescript-bot cherry-pick to release-5.4

**Pull request merged**